qiskit_ibm_runtime/qiskit_runtime_service.py

```diff
@@ -533,6 +596,37 @@ def backends(
                 ):
                     backends.append(backend)
 
+            if name and name not in self._backend_allowed_list:
+                instance_with_backend = None
+                # if instance is explicity passed in
+                if instance:
+                    if instance not in self._all_instances:
+                        raise IBMInputValueError(f"{instance} is not a valid instance.")
+                    if name not in self._backend_instance_groups.get(instance, []):
+                        raise IBMInputValueError(
+                            f"Backend {name} is not available in instance {instance}."
+                        )
+                    instance_with_backend = instance
+                else:  # if no instance passed in, check for default instance
+                    if not self._default_instance:
+                        # if no default instance set, check all instances and
+                        # fetch one that has the backend
+                        instance_with_backend = self._get_instance_from_grouping(name)
+
+                if instance_with_backend:
+                    # TODO how do we handle this logic here?
+                    # fetching a backend not in the current account instance requires a
+                    # different api client - this api client is tied to everything
+                    self._client_params.instance = instance_with_backend
+                    self._api_client = RuntimeClient(self._client_params)
```

Comment:
This code seems to be overwriting the api client. How would it behave if a user let's say instantiates the service, asks for backend 1 in instance A, asks for a backend 2 in instance B and decides to start a session with backend 1? Would this work?

Reply:
That's a good point, I'l have to test that edge case - I don't think that would work. I can't think of any other way to handle this since the backend is ultimately always tied to its instance.